TransIT.DAL/Repositories/BaseRepository.cs

```diff
@@ -17,53 +17,54 @@ public abstract class BaseRepository<TEntity> : IBaseRepository<TEntity>, IQuery
 
         public BaseRepository(TransITDBContext context)
         {
+            _entities = context.Set<TEntity>();
             _context = context;
         }
 
         public virtual Task<TEntity> GetByIdAsync(int id)
         {
-            return ComplexEntities.SingleOrDefaultAsync(t => t.Id.Equals(id));
+            return ComplexEntities.SingleOrDefaultAsync(entity => entity.Id == id);
         }
 
         public virtual Task<IEnumerable<TEntity>> GetAllAsync()
         {
-            return Task.FromResult<IEnumerable<TEntity>>(ComplexEntities);
+            return Task.FromResult(ComplexEntities.AsEnumerable());
```

Comment:
why did you remove tolistasync?

Reply:
Because i cannot convert Task<List<Employee>> to Task<IEnumerable<Employee>>